client/src/extension.ts

```diff
@@ -45,6 +45,12 @@ export async function activate (context: vscode.ExtensionContext): Promise<void>
       logger.loadSettings()
     }
   }))
+  context.subscriptions.push(vscode.workspace.onDidChangeWorkspaceFolders((event) => {

```

Comment:
So when a folder is added, we probably want to load the settings related to that specific folder if it has `workspaceFolder` level settings. They could be different across folders if users have set them up. The `getConfiguration` API optionally accepts a workspace folder as an argument.
When a folder is removed, we could dispose of the related resources if necessary.

Reply:
Adding a new folder doesn't trigger this. It's considered the same workspace. Opening a new workspace actually reloads the window and the extension as well. So this is only useful when opening vscode with no workspace, then opening a workspace.